drivers/volume/portworx/portworx.go

```diff
@@ -1462,6 +1463,78 @@ func (d *portworx) ValidatePureFaFbMountOptions(volumeName string, mountoption [
 
 }
 
+// ValidatePureFaCreateOptions validates FStype and createoptions with block size 2048 on those FStypes
+func (d *portworx) ValidatePureFaCreateOptions(volumeName string, FStype string, volumeNode *node.Node) error {
+	// Checking if file systems are properly set
+	FScmd := fmt.Sprintf(mountGrepVolume, volumeName)
+	FSout, err := d.nodeDriver.RunCommandWithNoRetry(
+		*volumeNode,
+		FScmd,
+		node.ConnectionOpts{
+			Timeout:         crashDriverTimeout,
+			TimeBeforeRetry: defaultRetryInterval,
+		})
+	if err != nil {
+		return fmt.Errorf("Failed to get mount response for volume %s, Err: %v", volumeName, err)
+	}
+	if strings.Contains(FSout, FStype) {
+		log.Infof("%s file system is available in the volume %s", FStype, volumeName)
+	} else {
+		return fmt.Errorf("Failed to get %s File system, Err: %v", FStype, err)
+	}
+
+	// Getting mapper volumename where createoptions are applied
+	mapperCmd := fmt.Sprintf(deviceMapper, volumeName)
+	mapperOut, err := d.nodeDriver.RunCommandWithNoRetry(
+		*volumeNode,
+		mapperCmd,
+		node.ConnectionOpts{
+			Timeout:         crashDriverTimeout,
+			TimeBeforeRetry: defaultRetryInterval,
+		})
+	if err != nil {
+		return fmt.Errorf("Failed to get attached volume for create option for pvc %s, Err: %v", volumeName, err)
+	}
+
+	// Validating implementation of create options
+	if FStype == "xfs" {
+		xfsInfoCmd := fmt.Sprintf("xfs_info %s ", strings.ReplaceAll(mapperOut, "\n", ""))
+		xfsInfoOut, err := d.nodeDriver.RunCommandWithNoRetry(
+			*volumeNode,
+			xfsInfoCmd,
+			node.ConnectionOpts{
+				Timeout:         crashDriverTimeout,
+				TimeBeforeRetry: defaultRetryInterval,
+			})
+		if err != nil {
+			return fmt.Errorf("Failed to get bsize for create option for pvc %s, Err: %v", volumeName, err)
+		}
+		if strings.Contains(xfsInfoOut, "bsize=2048") {
+			log.Infof("Blocksize 2048 is correctly configured by the create option of volume %s", volumeName)
+		} else {
+			return fmt.Errorf("Failed to get %s proper block size in the %s file system, Err: %v", xfsInfoOut, FStype, err)
+		}
+	} else if FStype == "ext4" {
+		ext4InfoCmd := fmt.Sprintf("tune2fs -l %s ", strings.ReplaceAll(mapperOut, "\n", ""))
+		ext4InfoOut, err := d.nodeDriver.RunCommandWithNoRetry(
+			*volumeNode,
+			ext4InfoCmd,
+			node.ConnectionOpts{
+				Timeout:         crashDriverTimeout,
+				TimeBeforeRetry: defaultRetryInterval,
+			})
+		if err != nil {
+			return fmt.Errorf("Failed to get bsize for create option for pvc %s, Err: %v", volumeName, err)
+		}
+		if strings.Contains(ext4InfoOut, "Block size:               2048") {
+			log.Infof("Blocksize 2048 is correctly configured by the create options of volume %s", volumeName)
+		} else {
+			return fmt.Errorf("Failed to get %s proper block size in the %s file system, Err: %v", ext4InfoOut, FStype, err)
+		}
+	}
```

Comment:
[nit] It might be good to add a warning here saying that this filesystem type isn't implemented? Just in case, not a requirement.

Reply:
Done.